src/kixi/datastore/metadatastore/events.clj

```diff
@@ -8,6 +8,31 @@
 (sh/alias 'user 'kixi.user)
 (sh/alias 'md 'kixi.datastore.metadatastore)
 
+(s/def ::spec-explain any?)
+
+;; Files
+
+(defmethod comms/event-payload
+  [:kixi.datastore/file-deleted "1.0.0"]
+  [_]
+  (s/keys :req [::md/id]))
+
+(sh/alias 'fd-reject 'kixi.event.file.delete.rejection)
+
+(s/def ::fd-reject/reason
+  #{:unauthorised
+    :incorrect-type
+    :invalid-cmd})
+
+(defmethod comms/event-payload
+  [:kixi.datastore/file-delete-rejected "1.0.0"]
+  [_]
+  (s/keys :req [::md/id]
+          :req-un [::fd-reject/reason]
+          :opt-un [::spec-explain]))
```

Comment:
Consider adding an 'and' to this to enforce the presence of the explain when the reason is :invalid

Reply:
Errr, could you help me with this?